src/fe-gtk/maingui.c

```diff
@@ -1119,6 +1145,21 @@ mg_tab_close (session *sess)
 	}
 }
 
+void
+mg_undo_tab_close (session *sess)
+{
+	closed_channel *chan;
+
+	if (g_queue_is_empty(&closed_channels))
+		return;
+
+	chan = g_queue_pop_head(&closed_channels);
+	if (is_server (chan->server))
```

Comment:
It's a little suboptimal to silently ignore the undo command if this is false. The user will have no idea why nothing happened, and if they press it again expecting it didn't go through the first time, it might reopen a completely different tab.